Exploration to reduce deadlocks

pushchris opened the pull request:
Simple change to potentially reduce gap locks by not having to re-select records for update

leobarcellos commented:
@pushchris I tried this PR but I also merged some other PR's (#585, #586, #587, #588) before updating.

On performance view, the queue number was getting higher and higher:

<img width="1479" alt="Screenshot 2024-12-25 at 5 02 23 PM" src="https://github.com/user-attachments/assets/322a2623-bdbf-4799-b24e-b2d4f499876d" />
<img width="1448" alt="Screenshot 2024-12-25 at 5 28 48 PM" src="https://github.com/user-attachments/assets/4f34e1fb-5ad8-465c-884e-ed916e6664f7" />

I tried modifying redis concurrency (10 -> 25) but with no success.

Then, I reverted our stack to the last PR before update (#581). Weirdly the queue task dropped to 3k, not sure what happened, queue throughput remained the same. (weird, maybe it was a bug on the performance view)

<img width="1454" alt="Screenshot 2024-12-25 at 5 49 19 PM" src="https://github.com/user-attachments/assets/ad317e81-3dd7-4bbb-a9e3-769be4f45116" />

Really not sure what happened.

Btw, I will increase servers capacity (ec2 and rds) later to see if it will mitigate some of these errors, ec2 and rds instance are floating around 80% cpu capacity

leobarcellos commented:
well, looks like it went back to 'normal'.
gonna try to update it back to the recent version

<img width="1445" alt="Screenshot 2024-12-25 at 6 54 25 PM" src="https://github.com/user-attachments/assets/078f7cf8-3d07-4176-9227-3e4116d88f82" />

leobarcellos commented:
At least for now the job throughput is normalized, it should process everything soon.

(I still don't know why the queue size went back to 1Mi+ the moment I updated to the recent version, maybe there is a bug indeed on the performance view)

<img width="1445" alt="Screenshot 2024-12-25 at 6 59 32 PM" src="https://github.com/user-attachments/assets/d53753c2-ebe0-40aa-8fdb-42b1b70ff217" />
<img width="1447" alt="Screenshot 2024-12-25 at 7 10 55 PM" src="https://github.com/user-attachments/assets/177266d3-b094-462b-a8df-df63181ed5d0" />

leobarcellos commented:
Well, today is even weirder. 3Mi tasks, sometimes with good throughput, somestimes with bad.

<img width="1453" alt="Screenshot 2024-12-26 at 4 14 33 PM" src="https://github.com/user-attachments/assets/3577448a-0053-4a5d-afc0-82f30ff3c247" />


I will revert it back again.

leobarcellos commented:
<img width="1472" alt="Screenshot 2024-12-26 at 4 32 22 PM" src="https://github.com/user-attachments/assets/76c31a0e-e5b2-47bf-8446-093d8e925a98" />

Reverted and boom, magically 3.7mi tasks in queue reduced to 28k -- throughput went back to normal as well

pushchris commented:
@leobarcellos my guess is that the list evaluation queue is literally never getting handled due to new priorities and your workers never getting to an empty queue state to handle. I've reverted that in this PR, the more I think about it the less having different priorities seems to be helpful and will mostly just cause issues. Can potentially re-evaluate moving everything to a lower priority queue and then selectively moving to high priority but even still that could cause lots of issues that are hard for folks to immediately identify. See if the latest commit to this PR stops the issue from happening for you

leobarcellos commented:
@pushchris Thanks!

Updated it ~20min ago, looks good to me:

<img width="1452" alt="Screenshot 2024-12-27 at 1 21 58 PM" src="https://github.com/user-attachments/assets/d371b11b-b3d7-4322-b158-fb8b3e53e863" />

--- by the way, last deadlock still was on dec 25 00h, its 60hrs+ without new deadlocks. But to be honest I don't know if it's 100% related with this PR, since I went back and forth with this PR and still hadn't a new deadlock. I think it was the parameter tweaking and the index update on journey_user_stat

pushchris commented:
> @pushchris Thanks!
> 
> Updated it ~20min ago, looks good to me:
> 
> <img alt="Screenshot 2024-12-27 at 1 21 58 PM" width="1452" src="https://private-user-images.githubusercontent.com/3453303/398973370-d371b11b-b3d7-4322-b158-fb8b3e53e863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0MjIzMTMsIm5iZiI6MTczNTQyMjAxMywicGF0aCI6Ii8zNDUzMzAzLzM5ODk3MzM3MC1kMzcxYjExYi1iM2Q3LTQzMjItYjE1OC1mYjhiM2U1M2U4NjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMjhUMjE0MDEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yjg1YjRhODhkNzAzYzA0YzM5NDE1NDVmZmJkZmUyMmMxZjliMzkzMjk0ZmJmM2IyMTc4N2Y2NDA1NzRkZDE3NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.k25TjAMt13tnpiBAIZ1PNyd2M1lIZVjik81-0j_uFqg">
> --- by the way, last deadlock still was on dec 25 00h, its 60hrs+ without new deadlocks. But to be honest I don't know if it's 100% related with this PR, since I went back and forth with this PR and still hadn't a new deadlock. I think it was the parameter tweaking and the index update on journey_user_stat

So bizarre on that index. In theory there is already an index on `journey_id` that the compiler can use so it shouldn't make a difference at all. Added it in to this PR and removed the merging bit. Would appreciate you giving it a whirl to see if any deadlocks come back. How many journeys do you have active in total?

pushchris commented:
@leobarcellos how have things been looking? Still no deadlocks since that addition?

leobarcellos commented:
@pushchris Hey there! Thanks for the update.

So, no more deadlocks found on journey_user_stat. Since then I received like 3 or 4 deadlock errors on campaign_sends, like these below. -- Which I think it's fine. I saw that you updated the failedStalled to update on chunks of 25, is the insertion doing in chunks as well?

About journeys, well, we do have a lot since there are many projects. Right now we have 165 journeys that are published and dont have the "deleted_at" column set. And the journey_user_step auto increment is at 47M and data length at 7.2G -- Would it be a good practice to have any sort of cleanup?


```
TRANSACTION 1962290683, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 7 lock struct(s), heap size 1128, 4 row lock(s)
MySQL thread id 572, OS thread handle 70398108542288, query id 343407676 172.30.6.37 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6081, '2024-12-31 12:00:00.000', 'pending', 432591), (6081, '2024-12-31 12:00:00.000', 'pending', 224919), (6081, '2024-12-31 12:00:00.000', 'pending', 355639), (6081, '2024-12-31 12:00:00.000', 'pending', 139211), (6081, '2024-12-31 12:00:00.000', 'pending', 459972), (6081, '2024-12-31 12:00:00.000', 'pending', 467801), (6081, '2024-12-31 12:00:00.000', 'pending', 103469), (6081, '2024-12-31 12:00:00.000', 'pending', 118135), (6081, '2024-12-31 12:00:00.000', 'pending', 467907), (6081, '2024-12-31 12:00:00.000', 'pending', 467848), (6081, '2024-12-31 12:00:00.000', 'pending', 301150), (6081, '2024-12-31 12:00:00.000', 'pending', 206261), (6081, '2024-12-31 12:00:00.000', 'pending', 108947), (6081, '2024-12-31 12:00:00.000', 'pending', 446684), (6081, '2024-12-31 12:00:00.000', 'pending', 468353), (6081, '2024-12-31 12:00:00.000', 'pending', 432605), (6081, '2024-12-31 12:00:00.000', 'pending', 459317), (6081, '2024-12-31 12:00:00.000', 'pending', 468309), (6081, '2024-12-31 12:00:00.000', 'pending', 128524), (6081, '2024-12-31 12:00:00.000', 'pending', 123728), (6081, '2024-12-31 12:00:00.000', 'pending', 135775), (6081, '2024-12-31 12:00:00.000', 'pending', 124162), (6081, '2024-12-31 12:00:00.000', 'pending', 224904), (6081, '2024-12-31 12:00:00.000', 'pending', 468527), (6081, '2024-12-31 12:00:00.000', 'pending', 453676), (6081, '2024-12-31 12:00:00.000', 'pending', 124042), (6081, '2024-12-31 12:00:00.000', 'pending', 452337), (6081, '2024-12-31 12:00:00.000', 'pending', 464192), (6081, '2024-12-31 12:00:00.000', 'pending', 466637), (6081, '2024-12-31 12:00:00.000', 'pending', 454985), (6081, '2024-12-31 12:00:00.000', 'pending', 469129), (6081, '2024-12-31 12:00:00.000', 'pending', 224927), (6081, '2024-12-31 12:00:00.000', 'pending', 224945), (6081, '2024-12-31 12:00:00.000', 'pending', 469637), (6081, '2024-12-31 12:00:00.000', 'pending', 467372), (6081, '2024-12-31 12:00:00.000', 'pending', 118347), (6081, '2024-12-31 12:00:00.000', 'pending', 207124), (6081, '2024-12-31 12:00:00.000', 'pending', 355637), (6081, '2024-12-31 12:00:00.000', 'pending', 108929), (6081, '2024-12-31 12:00:00.000', 'pending', 123812), (6081, '2024-12-31 12:00:00.000', 'pending', 136672), (6081, '2024-12-31 12:00:00.000', 'pending', 136584), (6081, '2024-12-31 12:00:00.000', 'pending', 128867), (6081, '2024-12-31 12:00:00.000', 'pending', 136201), (6081, '2024-12-31 12:00:00.000', 'pending', 103509), (6081, '2024-12-31 12:00:00.000', 'pending', 95433), (6081, '2024-12-31 12:00:00.000', 'pending', 260450), (6081, '2024-12-31 12:00:00.000', 'pending', 136332), (6081, '2024-12-31 12:00:00.000', 'pending', 136745), (6081, '2024-12-31 12:00:00.000', 'pending', 135761), (6081, '2024-12-31 12:00:00.000', 'pending', 207346), (6081, '2024-12-31 12:00:00.000', 'pending', 135745), (6081, '2024-12-31 12:00:00.000', 'pending', 136687), (6081, '2024-12-31 12:00:00.000', 'pending', 1087
RECORD LOCKS space id 49 page no 154162 n bits 328 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 1962290683 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 154162 n bits 328 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 1962290683 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 1962290684, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 7 lock struct(s), heap size 1128, 4 row lock(s)
MySQL thread id 755, OS thread handle 70370664527184, query id 343407677 172.30.7.141 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6082, '2025-01-01 12:00:00.000', 'pending', 103538), (6082, '2025-01-01 12:00:00.000', 'pending', 128551), (6082, '2025-01-01 12:00:00.000', 'pending', 124095), (6082, '2025-01-01 12:00:00.000', 'pending', 118020), (6082, '2025-01-01 12:00:00.000', 'pending', 123851), (6082, '2025-01-01 12:00:00.000', 'pending', 128593), (6082, '2025-01-01 12:00:00.000', 'pending', 128711), (6082, '2025-01-01 12:00:00.000', 'pending', 128970), (6082, '2025-01-01 12:00:00.000', 'pending', 103556), (6082, '2025-01-01 12:00:00.000', 'pending', 128844), (6082, '2025-01-01 12:00:00.000', 'pending', 124347), (6082, '2025-01-01 12:00:00.000', 'pending', 124165), (6082, '2025-01-01 12:00:00.000', 'pending', 124281), (6082, '2025-01-01 12:00:00.000', 'pending', 124040), (6082, '2025-01-01 12:00:00.000', 'pending', 136625), (6082, '2025-01-01 12:00:00.000', 'pending', 128909), (6082, '2025-01-01 12:00:00.000', 'pending', 128798), (6082, '2025-01-01 12:00:00.000', 'pending', 128929), (6082, '2025-01-01 12:00:00.000', 'pending', 128644), (6082, '2025-01-01 12:00:00.000', 'pending', 124322), (6082, '2025-01-01 12:00:00.000', 'pending', 68711), (6082, '2025-01-01 12:00:00.000', 'pending', 103647), (6082, '2025-01-01 12:00:00.000', 'pending', 543918), (6082, '2025-01-01 12:00:00.000', 'pending', 124421), (6082, '2025-01-01 12:00:00.000', 'pending', 124301), (6082, '2025-01-01 12:00:00.000', 'pending', 128652), (6082, '2025-01-01 12:00:00.000', 'pending', 124397), (6082, '2025-01-01 12:00:00.000', 'pending', 103431), (6082, '2025-01-01 12:00:00.000', 'pending', 578676), (6082, '2025-01-01 12:00:00.000', 'pending', 124270), (6082, '2025-01-01 12:00:00.000', 'pending', 124329), (6082, '2025-01-01 12:00:00.000', 'pending', 128507), (6082, '2025-01-01 12:00:00.000', 'pending', 124172), (6082, '2025-01-01 12:00:00.000', 'pending', 108766), (6082, '2025-01-01 12:00:00.000', 'pending', 118310), (6082, '2025-01-01 12:00:00.000', 'pending', 124220), (6082, '2025-01-01 12:00:00.000', 'pending', 128499), (6082, '2025-01-01 12:00:00.000', 'pending', 124382), (6082, '2025-01-01 12:00:00.000', 'pending', 123975), (6082, '2025-01-01 12:00:00.000', 'pending', 124263), (6082, '2025-01-01 12:00:00.000', 'pending', 128841), (6082, '2025-01-01 12:00:00.000', 'pending', 123726), (6082, '2025-01-01 12:00:00.000', 'pending', 124425), (6082, '2025-01-01 12:00:00.000', 'pending', 207422), (6082, '2025-01-01 12:00:00.000', 'pending', 108824), (6082, '2025-01-01 12:00:00.000', 'pending', 591551), (6082, '2025-01-01 12:00:00.000', 'pending', 123952), (6082, '2025-01-01 12:00:00.000', 'pending', 103381), (6082, '2025-01-01 12:00:00.000', 'pending', 103321), (6082, '2025-01-01 12:00:00.000', 'pending', 103411), (6082, '2025-01-01 12:00:00.000', 'pending', 136000), (6082, '2025-01-01 12:00:00.000', 'pending', 118639), (6082, '2025-01-01 12:00:00.000', 'pending', 124003), (6082, '2025-01-01 12:00:00.000', 'pending', 1243
RECORD LOCKS space id 49 page no 154162 n bits 328 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 1962290684 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 154162 n bits 328 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 1962290684 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
----------------------- END OF LOG ----------------------
```

pushchris commented:
Added one more tweak to hopefully better retry deadlocks in batch insert situations. Feel free to give this latest commit a whirl

leobarcellos commented:
I've updated our stack 3 hours ago -- I will update it here after 24h +-

By the way, this `['ER_LOCK_WAIT_TIMEOUT', 'ER_LOCK_DEADLOCK']` error treatment seems good, thank you!

leobarcellos commented:
No deadlocks since last update.
I will keep watching.

leobarcellos commented:
Unfortunately today we got some deadlocks. Not sure if big deal since it's being treated on the code. But I do realize it was like 50+ insertions (maybe even more since it truncate the query).

If we do these inserts on campaign_sends on smaller chunks, wouldn't it be better?

```
TRANSACTION 2305810811, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 9 lock struct(s), heap size 1128, 5 row lock(s)
MySQL thread id 5169, OS thread handle 70398478906704, query id 2121030283 172.30.8.73 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6481, '2025-01-08 09:00:00.000', 'pending', 326442), (6481, '2025-01-08 09:00:00.000', 'pending', 307478), (6481, '2025-01-08 09:00:00.000', 'pending', 378751), (6481, '2025-01-08 09:00:00.000', 'pending', 320306), (6481, '2025-01-08 09:00:00.000', 'pending', 360099), (6481, '2025-01-08 09:00:00.000', 'pending', 319983), (6481, '2025-01-08 09:00:00.000', 'pending', 320069), (6481, '2025-01-08 09:00:00.000', 'pending', 280502), (6481, '2025-01-08 09:00:00.000', 'pending', 161804), (6481, '2025-01-08 09:00:00.000', 'pending', 284214), (6481, '2025-01-08 09:00:00.000', 'pending', 325927), (6481, '2025-01-08 09:00:00.000', 'pending', 325485), (6481, '2025-01-08 09:00:00.000', 'pending', 309042), (6481, '2025-01-08 09:00:00.000', 'pending', 327680), (6481, '2025-01-08 09:00:00.000', 'pending', 337798), (6481, '2025-01-08 09:00:00.000', 'pending', 359915), (6481, '2025-01-08 09:00:00.000', 'pending', 335659), (6481, '2025-01-08 09:00:00.000', 'pending', 305473), (6481, '2025-01-08 09:00:00.000', 'pending', 369757), (6481, '2025-01-08 09:00:00.000', 'pending', 379824), (6481, '2025-01-08 09:00:00.000', 'pending', 276476), (6481, '2025-01-08 09:00:00.000', 'pending', 326573), (6481, '2025-01-08 09:00:00.000', 'pending', 317326), (6481, '2025-01-08 09:00:00.000', 'pending', 335524), (6481, '2025-01-08 09:00:00.000', 'pending', 161578), (6481, '2025-01-08 09:00:00.000', 'pending', 307568), (6481, '2025-01-08 09:00:00.000', 'pending', 333969), (6481, '2025-01-08 09:00:00.000', 'pending', 327788), (6481, '2025-01-08 09:00:00.000', 'pending', 322173), (6481, '2025-01-08 09:00:00.000', 'pending', 334057), (6481, '2025-01-08 09:00:00.000', 'pending', 380628), (6481, '2025-01-08 09:00:00.000', 'pending', 370127), (6481, '2025-01-08 09:00:00.000', 'pending', 380958), (6481, '2025-01-08 09:00:00.000', 'pending', 333818), (6481, '2025-01-08 09:00:00.000', 'pending', 317897), (6481, '2025-01-08 09:00:00.000', 'pending', 284530), (6481, '2025-01-08 09:00:00.000', 'pending', 335538), (6481, '2025-01-08 09:00:00.000', 'pending', 326000), (6481, '2025-01-08 09:00:00.000', 'pending', 319515), (6481, '2025-01-08 09:00:00.000', 'pending', 286516), (6481, '2025-01-08 09:00:00.000', 'pending', 307458), (6481, '2025-01-08 09:00:00.000', 'pending', 115658), (6481, '2025-01-08 09:00:00.000', 'pending', 380217), (6481, '2025-01-08 09:00:00.000', 'pending', 161541), (6481, '2025-01-08 09:00:00.000', 'pending', 333967), (6481, '2025-01-08 09:00:00.000', 'pending', 320193), (6481, '2025-01-08 09:00:00.000', 'pending', 326284), (6481, '2025-01-08 09:00:00.000', 'pending', 319463), (6481, '2025-01-08 09:00:00.000', 'pending', 345749), (6481, '2025-01-08 09:00:00.000', 'pending', 348619), (6481, '2025-01-08 09:00:00.000', 'pending', 380381), (6481, '2025-01-08 09:00:00.000', 'pending', 325366), (6481, '2025-01-08 09:00:00.000', 'pending', 325316), (6481, '2025-01-08 09:00:00.000', 'pending', 279
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305810811 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305810811 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 2305810817, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 8 lock struct(s), heap size 1128, 5 row lock(s)
MySQL thread id 5132, OS thread handle 70398644393296, query id 2121030296 172.30.7.22 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6479, '2025-01-07 20:00:00.000', 'pending', 664794), (6479, '2025-01-07 20:00:00.000', 'pending', 477062), (6479, '2025-01-07 20:00:00.000', 'pending', 631826), (6479, '2025-01-07 20:00:00.000', 'pending', 562143), (6479, '2025-01-07 20:00:00.000', 'pending', 677260), (6479, '2025-01-07 20:00:00.000', 'pending', 685478), (6479, '2025-01-07 20:00:00.000', 'pending', 685528), (6479, '2025-01-07 20:00:00.000', 'pending', 315653), (6479, '2025-01-07 20:00:00.000', 'pending', 527305), (6479, '2025-01-07 20:00:00.000', 'pending', 629956), (6479, '2025-01-07 20:00:00.000', 'pending', 542894), (6479, '2025-01-07 20:00:00.000', 'pending', 100808), (6479, '2025-01-07 20:00:00.000', 'pending', 664747), (6479, '2025-01-07 20:00:00.000', 'pending', 686116), (6479, '2025-01-07 20:00:00.000', 'pending', 653172), (6479, '2025-01-07 20:00:00.000', 'pending', 466898), (6479, '2025-01-07 20:00:00.000', 'pending', 660577), (6479, '2025-01-07 20:00:00.000', 'pending', 681176), (6479, '2025-01-07 20:00:00.000', 'pending', 686775), (6479, '2025-01-07 20:00:00.000', 'pending', 437083), (6479, '2025-01-07 20:00:00.000', 'pending', 686877), (6479, '2025-01-07 20:00:00.000', 'pending', 567378), (6479, '2025-01-07 20:00:00.000', 'pending', 679298), (6479, '2025-01-07 20:00:00.000', 'pending', 513094), (6479, '2025-01-07 20:00:00.000', 'pending', 508864), (6479, '2025-01-07 20:00:00.000', 'pending', 438142), (6479, '2025-01-07 20:00:00.000', 'pending', 586237), (6479, '2025-01-07 20:00:00.000', 'pending', 677857), (6479, '2025-01-07 20:00:00.000', 'pending', 687599), (6479, '2025-01-07 20:00:00.000', 'pending', 667094), (6479, '2025-01-07 20:00:00.000', 'pending', 687658), (6479, '2025-01-07 20:00:00.000', 'pending', 364539), (6479, '2025-01-07 20:00:00.000', 'pending', 674313), (6479, '2025-01-07 20:00:00.000', 'pending', 687903), (6479, '2025-01-07 20:00:00.000', 'pending', 687859), (6479, '2025-01-07 20:00:00.000', 'pending', 688045), (6479, '2025-01-07 20:00:00.000', 'pending', 360669), (6479, '2025-01-07 20:00:00.000', 'pending', 667888), (6479, '2025-01-07 20:00:00.000', 'pending', 688160), (6479, '2025-01-07 20:00:00.000', 'pending', 476836), (6479, '2025-01-07 20:00:00.000', 'pending', 688232), (6479, '2025-01-07 20:00:00.000', 'pending', 387805), (6479, '2025-01-07 20:00:00.000', 'pending', 456364), (6479, '2025-01-07 20:00:00.000', 'pending', 551358), (6479, '2025-01-07 20:00:00.000', 'pending', 432836), (6479, '2025-01-07 20:00:00.000', 'pending', 392313), (6479, '2025-01-07 20:00:00.000', 'pending', 664421), (6479, '2025-01-07 20:00:00.000', 'pending', 636919), (6479, '2025-01-07 20:00:00.000', 'pending', 403085), (6479, '2025-01-07 20:00:00.000', 'pending', 679149), (6479, '2025-01-07 20:00:00.000', 'pending', 677968), (6479, '2025-01-07 20:00:00.000', 'pending', 688294), (6479, '2025-01-07 20:00:00.000', 'pending', 558213), (6479, '2025-01-07 20:00:00.000', 'pending', 486
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305810817 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305810817 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 2305816111, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 11 lock struct(s), heap size 1128, 7 row lock(s)
MySQL thread id 5149, OS thread handle 70398549452112, query id 2121047134 172.30.7.22 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6480, '2025-01-06 20:00:00.000', 'pending', 322418), (6480, '2025-01-06 20:00:00.000', 'pending', 316170), (6480, '2025-01-06 20:00:00.000', 'pending', 321842), (6480, '2025-01-06 20:00:00.000', 'pending', 302472), (6480, '2025-01-06 20:00:00.000', 'pending', 309476), (6480, '2025-01-06 20:00:00.000', 'pending', 322917), (6480, '2025-01-06 20:00:00.000', 'pending', 323105), (6480, '2025-01-06 20:00:00.000', 'pending', 309801), (6480, '2025-01-06 20:00:00.000', 'pending', 323087), (6480, '2025-01-06 20:00:00.000', 'pending', 319470), (6480, '2025-01-06 20:00:00.000', 'pending', 285967), (6480, '2025-01-06 20:00:00.000', 'pending', 323199), (6480, '2025-01-06 20:00:00.000', 'pending', 323322), (6480, '2025-01-06 20:00:00.000', 'pending', 305112), (6480, '2025-01-06 20:00:00.000', 'pending', 302463), (6480, '2025-01-06 20:00:00.000', 'pending', 323148), (6480, '2025-01-06 20:00:00.000', 'pending', 315453), (6480, '2025-01-06 20:00:00.000', 'pending', 322635), (6480, '2025-01-06 20:00:00.000', 'pending', 320340), (6480, '2025-01-06 20:00:00.000', 'pending', 307345), (6480, '2025-01-06 20:00:00.000', 'pending', 320508), (6480, '2025-01-06 20:00:00.000', 'pending', 320531), (6480, '2025-01-06 20:00:00.000', 'pending', 242703), (6480, '2025-01-06 20:00:00.000', 'pending', 322282), (6480, '2025-01-06 20:00:00.000', 'pending', 307779), (6480, '2025-01-06 20:00:00.000', 'pending', 320938), (6480, '2025-01-06 20:00:00.000', 'pending', 319431), (6480, '2025-01-06 20:00:00.000', 'pending', 305148), (6480, '2025-01-06 20:00:00.000', 'pending', 322506), (6480, '2025-01-06 20:00:00.000', 'pending', 308924), (6480, '2025-01-06 20:00:00.000', 'pending', 297857), (6480, '2025-01-06 20:00:00.000', 'pending', 323139), (6480, '2025-01-06 20:00:00.000', 'pending', 323158), (6480, '2025-01-06 20:00:00.000', 'pending', 309677), (6480, '2025-01-06 20:00:00.000', 'pending', 325238), (6480, '2025-01-06 20:00:00.000', 'pending', 326322), (6480, '2025-01-06 20:00:00.000', 'pending', 326572), (6480, '2025-01-06 20:00:00.000', 'pending', 320475), (6480, '2025-01-06 20:00:00.000', 'pending', 322680), (6480, '2025-01-06 20:00:00.000', 'pending', 327700), (6480, '2025-01-06 20:00:00.000', 'pending', 326150), (6480, '2025-01-06 20:00:00.000', 'pending', 327765), (6480, '2025-01-06 20:00:00.000', 'pending', 320933), (6480, '2025-01-06 20:00:00.000', 'pending', 302450), (6480, '2025-01-06 20:00:00.000', 'pending', 316318), (6480, '2025-01-06 20:00:00.000', 'pending', 329353), (6480, '2025-01-06 20:00:00.000', 'pending', 329578), (6480, '2025-01-06 20:00:00.000', 'pending', 324984), (6480, '2025-01-06 20:00:00.000', 'pending', 329690), (6480, '2025-01-06 20:00:00.000', 'pending', 302444), (6480, '2025-01-06 20:00:00.000', 'pending', 328964), (6480, '2025-01-06 20:00:00.000', 'pending', 325010), (6480, '2025-01-06 20:00:00.000', 'pending', 325011), (6480, '2025-01-06 20:00:00.000', 'pending', 325
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305816111 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305816111 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 2305816121, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 8 lock struct(s), heap size 1128, 5 row lock(s)
MySQL thread id 5191, OS thread handle 70391799711056, query id 2121047160 172.30.6.37 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6481, '2025-01-08 09:00:00.000', 'pending', 309084), (6481, '2025-01-08 09:00:00.000', 'pending', 345321), (6481, '2025-01-08 09:00:00.000', 'pending', 101159), (6481, '2025-01-08 09:00:00.000', 'pending', 344857), (6481, '2025-01-08 09:00:00.000', 'pending', 319545), (6481, '2025-01-08 09:00:00.000', 'pending', 152558), (6481, '2025-01-08 09:00:00.000', 'pending', 96361), (6481, '2025-01-08 09:00:00.000', 'pending', 344439), (6481, '2025-01-08 09:00:00.000', 'pending', 344667), (6481, '2025-01-08 09:00:00.000', 'pending', 280220), (6481, '2025-01-08 09:00:00.000', 'pending', 345845), (6481, '2025-01-08 09:00:00.000', 'pending', 333601), (6481, '2025-01-08 09:00:00.000', 'pending', 334047), (6481, '2025-01-08 09:00:00.000', 'pending', 307572), (6481, '2025-01-08 09:00:00.000', 'pending', 319906), (6481, '2025-01-08 09:00:00.000', 'pending', 319606), (6481, '2025-01-08 09:00:00.000', 'pending', 307357), (6481, '2025-01-08 09:00:00.000', 'pending', 369470), (6481, '2025-01-08 09:00:00.000', 'pending', 327770), (6481, '2025-01-08 09:00:00.000', 'pending', 325161), (6481, '2025-01-08 09:00:00.000', 'pending', 317135), (6481, '2025-01-08 09:00:00.000', 'pending', 344873), (6481, '2025-01-08 09:00:00.000', 'pending', 315538), (6481, '2025-01-08 09:00:00.000', 'pending', 363093), (6481, '2025-01-08 09:00:00.000', 'pending', 362436), (6481, '2025-01-08 09:00:00.000', 'pending', 321864), (6481, '2025-01-08 09:00:00.000', 'pending', 326119), (6481, '2025-01-08 09:00:00.000', 'pending', 345142), (6481, '2025-01-08 09:00:00.000', 'pending', 360951), (6481, '2025-01-08 09:00:00.000', 'pending', 320229), (6481, '2025-01-08 09:00:00.000', 'pending', 369997), (6481, '2025-01-08 09:00:00.000', 'pending', 320263), (6481, '2025-01-08 09:00:00.000', 'pending', 360168), (6481, '2025-01-08 09:00:00.000', 'pending', 335178), (6481, '2025-01-08 09:00:00.000', 'pending', 317914), (6481, '2025-01-08 09:00:00.000', 'pending', 309055), (6481, '2025-01-08 09:00:00.000', 'pending', 335274), (6481, '2025-01-08 09:00:00.000', 'pending', 361840), (6481, '2025-01-08 09:00:00.000', 'pending', 335407), (6481, '2025-01-08 09:00:00.000', 'pending', 326571), (6481, '2025-01-08 09:00:00.000', 'pending', 347878), (6481, '2025-01-08 09:00:00.000', 'pending', 327880), (6481, '2025-01-08 09:00:00.000', 'pending', 348220), (6481, '2025-01-08 09:00:00.000', 'pending', 371550), (6481, '2025-01-08 09:00:00.000', 'pending', 88416), (6481, '2025-01-08 09:00:00.000', 'pending', 94029), (6481, '2025-01-08 09:00:00.000', 'pending', 321738), (6481, '2025-01-08 09:00:00.000', 'pending', 348601), (6481, '2025-01-08 09:00:00.000', 'pending', 304723), (6481, '2025-01-08 09:00:00.000', 'pending', 371464), (6481, '2025-01-08 09:00:00.000', 'pending', 348794), (6481, '2025-01-08 09:00:00.000', 'pending', 348825), (6481, '2025-01-08 09:00:00.000', 'pending', 325059), (6481, '2025-01-08 09:00:00.000', 'pending', 336258
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305816121 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305816121 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 2305837957, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 8 lock struct(s), heap size 1128, 4 row lock(s)
MySQL thread id 5087, OS thread handle 70398063232336, query id 2121133887 172.30.7.141 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6480, '2025-01-06 20:00:00.000', 'pending', 560910), (6480, '2025-01-06 20:00:00.000', 'pending', 531519), (6480, '2025-01-06 20:00:00.000', 'pending', 518789), (6480, '2025-01-06 20:00:00.000', 'pending', 562442), (6480, '2025-01-06 20:00:00.000', 'pending', 527847), (6480, '2025-01-06 20:00:00.000', 'pending', 563281), (6480, '2025-01-06 20:00:00.000', 'pending', 551384), (6480, '2025-01-06 20:00:00.000', 'pending', 518489), (6480, '2025-01-06 20:00:00.000', 'pending', 549165), (6480, '2025-01-06 20:00:00.000', 'pending', 557756), (6480, '2025-01-06 20:00:00.000', 'pending', 537359), (6480, '2025-01-06 20:00:00.000', 'pending', 511028), (6480, '2025-01-06 20:00:00.000', 'pending', 310728), (6480, '2025-01-06 20:00:00.000', 'pending', 563174), (6480, '2025-01-06 20:00:00.000', 'pending', 514768), (6480, '2025-01-06 20:00:00.000', 'pending', 567930), (6480, '2025-01-06 20:00:00.000', 'pending', 567971), (6480, '2025-01-06 20:00:00.000', 'pending', 563805), (6480, '2025-01-06 20:00:00.000', 'pending', 567877), (6480, '2025-01-06 20:00:00.000', 'pending', 568485), (6480, '2025-01-06 20:00:00.000', 'pending', 568011), (6480, '2025-01-06 20:00:00.000', 'pending', 469372), (6480, '2025-01-06 20:00:00.000', 'pending', 558982), (6480, '2025-01-06 20:00:00.000', 'pending', 569286), (6480, '2025-01-06 20:00:00.000', 'pending', 569362), (6480, '2025-01-06 20:00:00.000', 'pending', 498325), (6480, '2025-01-06 20:00:00.000', 'pending', 473332), (6480, '2025-01-06 20:00:00.000', 'pending', 557182), (6480, '2025-01-06 20:00:00.000', 'pending', 548484), (6480, '2025-01-06 20:00:00.000', 'pending', 460258), (6480, '2025-01-06 20:00:00.000', 'pending', 499317), (6480, '2025-01-06 20:00:00.000', 'pending', 561999), (6480, '2025-01-06 20:00:00.000', 'pending', 559715), (6480, '2025-01-06 20:00:00.000', 'pending', 568536), (6480, '2025-01-06 20:00:00.000', 'pending', 551466), (6480, '2025-01-06 20:00:00.000', 'pending', 567799), (6480, '2025-01-06 20:00:00.000', 'pending', 310396), (6480, '2025-01-06 20:00:00.000', 'pending', 485120), (6480, '2025-01-06 20:00:00.000', 'pending', 544591), (6480, '2025-01-06 20:00:00.000', 'pending', 563659), (6480, '2025-01-06 20:00:00.000', 'pending', 549292), (6480, '2025-01-06 20:00:00.000', 'pending', 560980), (6480, '2025-01-06 20:00:00.000', 'pending', 570950), (6480, '2025-01-06 20:00:00.000', 'pending', 532256), (6480, '2025-01-06 20:00:00.000', 'pending', 359809), (6480, '2025-01-06 20:00:00.000', 'pending', 563607), (6480, '2025-01-06 20:00:00.000', 'pending', 310187), (6480, '2025-01-06 20:00:00.000', 'pending', 485890), (6480, '2025-01-06 20:00:00.000', 'pending', 565072), (6480, '2025-01-06 20:00:00.000', 'pending', 555725), (6480, '2025-01-06 20:00:00.000', 'pending', 552443), (6480, '2025-01-06 20:00:00.000', 'pending', 518135), (6480, '2025-01-06 20:00:00.000', 'pending', 582114), (6480, '2025-01-06 20:00:00.000', 'pending', 368
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305837957 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305837957 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
TRANSACTION 2305837958, ACTIVE 0 sec inserting
mysql tables in use 1, locked 1
LOCK WAIT 7 lock struct(s), heap size 1128, 4 row lock(s)
MySQL thread id 5086, OS thread handle 70398649955664, query id 2121133888 172.30.8.73 overwatch update
insert ignore into `campaign_sends` (`campaign_id`, `send_at`, `state`, `user_id`) values (6481, '2025-01-08 09:00:00.000', 'pending', 544129), (6481, '2025-01-08 09:00:00.000', 'pending', 540951), (6481, '2025-01-08 09:00:00.000', 'pending', 520184), (6481, '2025-01-08 09:00:00.000', 'pending', 466945), (6481, '2025-01-08 09:00:00.000', 'pending', 537650), (6481, '2025-01-08 09:00:00.000', 'pending', 345208), (6481, '2025-01-08 09:00:00.000', 'pending', 527850), (6481, '2025-01-08 09:00:00.000', 'pending', 391986), (6481, '2025-01-08 09:00:00.000', 'pending', 325645), (6481, '2025-01-08 09:00:00.000', 'pending', 319512), (6481, '2025-01-08 09:00:00.000', 'pending', 455840), (6481, '2025-01-08 09:00:00.000', 'pending', 541411), (6481, '2025-01-08 09:00:00.000', 'pending', 547786), (6481, '2025-01-08 09:00:00.000', 'pending', 549748), (6481, '2025-01-08 09:00:00.000', 'pending', 536070), (6481, '2025-01-08 09:00:00.000', 'pending', 513796), (6481, '2025-01-08 09:00:00.000', 'pending', 549333), (6481, '2025-01-08 09:00:00.000', 'pending', 397003), (6481, '2025-01-08 09:00:00.000', 'pending', 451227), (6481, '2025-01-08 09:00:00.000', 'pending', 549756), (6481, '2025-01-08 09:00:00.000', 'pending', 470737), (6481, '2025-01-08 09:00:00.000', 'pending', 326427), (6481, '2025-01-08 09:00:00.000', 'pending', 472091), (6481, '2025-01-08 09:00:00.000', 'pending', 522764), (6481, '2025-01-08 09:00:00.000', 'pending', 532186), (6481, '2025-01-08 09:00:00.000', 'pending', 543400), (6481, '2025-01-08 09:00:00.000', 'pending', 550886), (6481, '2025-01-08 09:00:00.000', 'pending', 551278), (6481, '2025-01-08 09:00:00.000', 'pending', 540433), (6481, '2025-01-08 09:00:00.000', 'pending', 551417), (6481, '2025-01-08 09:00:00.000', 'pending', 551120), (6481, '2025-01-08 09:00:00.000', 'pending', 528557), (6481, '2025-01-08 09:00:00.000', 'pending', 547494), (6481, '2025-01-08 09:00:00.000', 'pending', 533111), (6481, '2025-01-08 09:00:00.000', 'pending', 551183), (6481, '2025-01-08 09:00:00.000', 'pending', 551080), (6481, '2025-01-08 09:00:00.000', 'pending', 549500), (6481, '2025-01-08 09:00:00.000', 'pending', 536722), (6481, '2025-01-08 09:00:00.000', 'pending', 543902), (6481, '2025-01-08 09:00:00.000', 'pending', 551825), (6481, '2025-01-08 09:00:00.000', 'pending', 522482), (6481, '2025-01-08 09:00:00.000', 'pending', 538026), (6481, '2025-01-08 09:00:00.000', 'pending', 528786), (6481, '2025-01-08 09:00:00.000', 'pending', 542894), (6481, '2025-01-08 09:00:00.000', 'pending', 542828), (6481, '2025-01-08 09:00:00.000', 'pending', 550315), (6481, '2025-01-08 09:00:00.000', 'pending', 545164), (6481, '2025-01-08 09:00:00.000', 'pending', 518845), (6481, '2025-01-08 09:00:00.000', 'pending', 546388), (6481, '2025-01-08 09:00:00.000', 'pending', 547746), (6481, '2025-01-08 09:00:00.000', 'pending', 552854), (6481, '2025-01-08 09:00:00.000', 'pending', 392560), (6481, '2025-01-08 09:00:00.000', 'pending', 550000), (6481, '2025-01-08 09:00:00.000', 'pending', 543
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305837958 lock_mode X
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
RECORD LOCKS space id 49 page no 184899 n bits 168 index PRIMARY of table `parcelvoy`.`campaign_sends` trx id 2305837958 lock_mode X insert intention waiting
Record lock, heap no 1 PHYSICAL RECORD: n_fields 1; compact format; info bits 0
0: len 8; hex 73757072656d756d; asc supremum;;
----------------------- END OF LOG ----------------------
```
